pennylane/transforms/split_qscript.py

```diff
@@ -386,20 +282,41 @@ def _group_measurements(measurements: List[MeasurementProcess]) -> List[List[Mea
     Returns:
         List[List[MeasurementProcess]]: list of groups of observables with non overlapping wires
     """
-    qwc_groups = []
+    qwc_groups = []  # [(wires (Wires), m_group (list), is_pauli (bool))]
     for m in measurements:
         if len(m.wires) == 0:  # measurement acts on all wires: e.g. qml.counts()
-            qwc_groups.append((m.wires, [m]))
+            qwc_groups.append((m.wires, [m], True))  # use True because observable is ``qml.PauliZ``
             continue
 
         op_added = False
-        for idx, (wires, group) in enumerate(qwc_groups):
+        # when m.obs is None the measurement acts on the basis states, and thus we can assume that
+        # the observable is ``qml.PauliZ`` for all measurement wires
+        obs = m.obs or _pauli_z(m.wires)
+        for idx, (wires, group, is_pauli) in enumerate(qwc_groups):
+            if is_pauli and is_pauli_word(obs):
+                # use pauli grouping functionality if all the elements in the group are pauli words
+                obs_group = [m.obs or _pauli_z(m.wires) for m in group]
+                if len(group_observables(obs_group + [obs])) == 1:
+                    qwc_groups[idx] = (wires + m.wires, group + [m], True)
+                    op_added = True
+                    break
+            # check overlapping wires
             if len(wires) > 0 and all(wire not in m.wires for wire in wires):
-                qwc_groups[idx] = (wires + m.wires, group + [m])
+                qwc_groups[idx] = (wires + m.wires, group + [m], False)
                 op_added = True
                 break
 
         if not op_added:
-            qwc_groups.append((m.wires, [m]))
+            qwc_groups.append((m.wires, [m], is_pauli_word(obs)))
 
     return [group[1] for group in qwc_groups]
+
+
+def _pauli_z(wires: Wires):
+    """Generate ``PauliZ`` operator.
+
+    Args:
+        wires (Wires): wires that the operator acts on"""
+    if len(wires) == 1:
+        return qml.PauliZ(0)
+    return Tensor(*[qml.PauliZ(w) for w in wires])
```

Comment:
Do we have observables that measure multiple wires?

Reply:
Yes! For example: `qml.probs(wires=[0, 1])`